cou/utils/juju_utils.py

```diff
@@ -232,20 +241,22 @@ async def async_get_unit_from_name(unit_name, model=None, model_name=None):
         if model is None:
             model = await _async_get_model(model_name)
         units = model.applications[app].units
-    except KeyError:
-        msg = "Application: {} does not exist in current model".format(app)
+    except KeyError as exc:
+        msg = f"Application: {app} does not exist in current model"
         logger.error(msg)
-        raise UnitNotFound(unit_name)
-    for u in units:
-        if u.entity_id == unit_name:
-            unit = u
+        raise UnitNotFound(unit_name) from exc
+    for single_unit in units:
+        if single_unit.entity_id == unit_name:
+            unit = single_unit
             break
     else:
         raise UnitNotFound(unit_name)
     return unit
 
 
-async def async_get_application_config(application_name, model_name=None):
+async def async_get_application_config(
+    application_name: str, model_name: Optional[str] = None
+) -> Application:
     """Return application configuration.
 
     :param model_name: Name of model to query.
```

Comment:
change `:type model_name: str` to `:type model_name: Optional[str]`

Reply:
ok